### PR TITLE
Implementation of Space Traveller's Hub Project

### PR DESCRIPTION
**Pull request by Reem-lab**
## Requirements of the Project that we Made here 👍 💯🚀🚀
 - Set Up the project with the Kanban board to Divide the Tasks between the team
 - Set up with GitFlow.
 - Create React App.
 - Install [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://reactrouter.com/web/guides/quick-start).
 - Download the image for the app logo.🚀
 - Create routes and view components (Rockets, Missions, My Profile, Dragons).
 - Create directories for all Redux state slice files (rockets, missions, dragons.📂
 - Fetch data and update Redux store. ♻
 - Dispatch an action to store the selected data in Redux store.
 - Render the list of `Rockets` & `Missions` & `Dragons`.
 - Implement the `Rockets.js` component , `missions.js` and `dragons`.
 - Implement the button of Rocket to toggle `Reservation` and `Cancelation`.
 - Implement the badge to the `reserved` rocket.
 - Implement the button to `Join` and `leave` mission
 - Implement the button of `Reserve` and `Cancel` Dragon.
 - Implement the filter to the reserved `Rocket` &`Mission` & `Dragon`.
 - Add styles to the three pages
 - No linters error.💯💯🔥🔥💪

### CODE REVIEW

File: src/redux/actions/missionsActions.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+export const setFetchedMissions = (missions) => ({
+  type: SET_MISSIONS,
+  missions,
+});
+
+export const fetchMissions = () => (dispatch) => {
+  fetch('https://api.spacexdata.com/v3/missions')
+    .then((response) => response.json())
+    .then((data) => dispatch(setFetchedMissions(data)));
```
Comment (TedLivist) [inline src/redux/actions/missionsActions.js]: - [x] Currently, upon fetching the data, your state stores unnecessary data. As stated in the requirements only the selected API data (`mission_id`, `mission_name`, `description`) should be dispatched to the store. Hence you should ensure the data from the API is filtered and only the correct ones are saved to the store. This is correctly done in the  `src/redux/actions/RocketsActions.js` file.

  Data to dispatch | Your dispatched data
  :---:|:---:
  reem-lab-missions-data | reem-lab-missions

File: src/redux/actions/dragonsActions.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+export const toggleDragonReservation = (id) => ({
+  type: TOGGLE_DRAGON_RESERVATION,
+  id,
+});
+
+export const fetchDragons = () => (dispatch) => {
+  fetch(basicURL)
+    .then((resolve) => resolve.json())
+    .then((data) => {
+      dispatch(displayFetchedDragons(data));
```
Comment (TedLivist) [inline src/redux/actions/dragonsActions.js]: - [x] Upon fetching the data kindly ensure that only the selected data (`id`, `name`, `type`, `flickr_images`) is dispatched to the store. Right now you have unnecessary data in your store. Kindly filter for only the needed data.

  Data to dispatch | Your dispatched data
  :---:|:---:
  reem-lab-dragons-data | reem-lab-dragons

File: src/components/MyProfile.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        ))}
+      </div>
+      <div className="my-profile__dragons">
+        <h2>My Dragons</h2>
+        {myDragons.map((dragon) => (
+          dragon.reserved && (
+            <div key={dragon.id}>
+              <p>{dragon.name}</p>
+            </div>
+          )
```
Comment (TedLivist) [inline src/components/MyProfile.js]: - [x] Your reserved Rockets, Missions, and Dragons should be rendered using the `filter` method. Kindly do that.

  **SUGGESTION** You can consider filtering the `myMissions`, `myDragons`, `rockets` variables before mapping. 

  reem-lab-filter

File: src/tests/App.test.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from '@testing-library/react';
+import App from '../App';
+
+test('renders learn react link', () => {
+  render(<App />);
+  const linkElement = screen.getByText(/learn react/i);
+  expect(linkElement).toBeInTheDocument();
+});
```
Comment (TedLivist) [inline src/tests/App.test.js]: - [x] You currently have only one test whereas you should have more tests. Kindly add more tests for your app components.

  reem-lab-tests-2

- [x] Your test is failing. Kindly fix it.

  reem-lab-tests

File: src/components/Rockets.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+              <img className="img" src={rocket.flickr_images} alt={rocket.rocket_name} />
+            </div>
+            <div className="info">
+              <h2>{rocket.rocket_name}</h2>
+              <p className="para">
+                <span className={(rocket.reserved ? 'reserved' : 'cancel')}>{(rocket.reserved ? 'Reserved' : ' ') }</span>
+                {rocket.description}
+              </p>
+              <button className={(!rocket.reserved ? 'reserveBtn' : 'cancelBtn')} type="button" onClick={() => toggleReservedBtn(rocket.id)}>{(!rocket.reserved ? 'Reserve Rockets' : 'Cancel Reservation') }</button>
+            </div>
```
Comment (TedLivist) [inline src/components/Rockets.js]: - [x] **[OPTIONAL]** In order to make your **Rockets** component look neater, I would recommend you extract this code into a reusable component responsible for a single **Rocket** and render that component here instead.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi, @Reem-lab and @fabgrel10 and @eduardosancho,

Good job so far!
✔️ Good documentation
✔️ Reservations are rightly toggled upon user click
✔️ Data is correctly fetched
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi Team 👋🏽  

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

**Pull request merged**